network/p2p/p2pnode/libp2pStream_test.go

```diff
@@ -421,103 +428,33 @@ func TestUnicastOverStream_Fallback(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	signalerCtx := irrecoverable.NewMockSignalerContext(t, ctx)
 
-	ch := make(chan string, 2) // we expect two messages during test, one from node1->node2 and vice versa.
-
-	// Create the handler function
-	streamHandler := func(s network.Stream) {
-		rw := bufio.NewReadWriter(bufio.NewReader(s), bufio.NewWriter(s))
-		str, err := rw.ReadString('\n')
-		require.NoError(t, err)
-		ch <- str
-	}
-
 	// Creates nodes
 	// node1: supports only plain unicast protocol
 	// node2: supports plain and gzip
 	sporkId := unittest.IdentifierFixture()
-	node1, id1 := p2pfixtures.NodeFixture(
+
+	streamHandler1, inbound1 := p2pfixtures.StreamHandlerFixture(t)
+	node1, _ := p2pfixtures.NodeFixture(
 		t,
 		sporkId,
-		"test_unicast_over_stream_fallback",
-		p2pfixtures.WithDefaultStreamHandler(streamHandler),
+		t.Name(),
+		p2pfixtures.WithDefaultStreamHandler(streamHandler1),
 	)
 
-	node2, id2 := p2pfixtures.NodeFixture(
+	streamHandler2, inbound2 := p2pfixtures.StreamHandlerFixture(t)
+	node2, _ := p2pfixtures.NodeFixture(
 		t,
 		sporkId,
-		"test_unicast_over_stream_fallback",
-		p2pfixtures.WithDefaultStreamHandler(streamHandler),
+		t.Name(),
+		p2pfixtures.WithDefaultStreamHandler(streamHandler2),
 		p2pfixtures.WithPreferredUnicasts([]unicast.ProtocolName{unicast.GzipCompressionUnicast}),
 	)
 
 	nodes := []p2p.LibP2PNode{node1, node2}
 	p2pfixtures.StartNodes(t, signalerCtx, nodes, 100*time.Millisecond)
 	defer p2pfixtures.StopNodes(t, nodes, cancel, 100*time.Millisecond)
 
-	testUnicastOverStreamRoundTrip(t, ctx, id1, node1, id2, node2, ch)
-}
-
-// testUnicastOverStreamRoundTrip checks node1 and node2 can create stream between each other and push unicast messages
-// to each other over the streams.
-//
-// The channel argument keeps the individual messages received at both ends.
-func testUnicastOverStreamRoundTrip(t *testing.T,
```

Comment:
Reason for removal: this test helper logic has been moved to fixtures, and generalized and revamped as `EnsureMessageExchangeOverUnicast` (see [here](https://github.com/onflow/flow-go/blob/af3662ff05c5c496d4edf69ef838080fbc354600/network/internal/p2pfixtures/fixtures.go#L609))